### ISSUE TITLE
About the structure of configuration files.

### ISSUE THREAD

**Issue by sakurawald** [fuji-fabric/fuji#88]
In the history of fuji, all the module data are store in `config.json`, including the `module enable supplier` and `module behaviour controller`, which makes the `config.json` a big file.

I am considering to flatten the structure of `config.json`, explicitly split the `module enable supplier` and the `module behaviour controller`.

Take module `chat.history` for example, you can enable this module in `config/fuji/config.json`, and modify the behaviour control in `config/fuji/modules/chat/config.json`.

Here are the problems:
1. There will be many files inside `config/fuji/`, which looks scary.
2. Most of the file are small files.

The benefits we get are:
1. The `config.json` file is only used to switch the `module enable supplier`.
2. The behaviour of module can be splitted into standalone file, makes it more compatible with other mods. (Yes, put all configurations inside one file, means the jvm will load the releavant symbols, which has a small possibility to crash the server, if it's not compatible with one of other mods.)
3. Easier to configure the modules, and less possbility to break the configuration file.

In short, in the past, the `config.json` file reveals the `physical structure` of the internal codes, and now, we flatten the `config.json` file into many small `config.<module-path>.json` file, and leaves a `logic structure` in `config.json`.
